pinc/upload_file.inc

```diff
@@ -316,7 +316,7 @@ function is_valid_dirname($dirname)
     }
 
     // We allow directory names to start with an alphanumeric, followed by 0
-    // more alphanumerics, underscores, dashes, and periods. The directory
+    // or more alphanumerics, underscores, dashes, and periods. The directory
     // name is valid if the regexp matches exactly once.
-    return preg_match('/^\w[\w-\.]*$/', $dirname) == 1;
+    return preg_match('/^\w[\w\.-]*$/', $dirname) == 1;
```

Comment:
Should we escape the hyphen to avoid accidental ranges like here?

Reply:
We can. If we do we should escape the one in the function above this too.

Reply:
Done.